conftest read json file failed due to json file contains BOM

Hi,

We met some problems when using conftest. If the json file contains BOM, conftest will throw `Error: running test: get configurations: get configurations: parser unmarshal: unmarshal json: invalid character 'ï' looking for beginning of value` error.
After removing them, conftest works.
So I think conftest need to support it, and could you fix this problem?

Your sample appears to include extra bytes, I suspect they're unicode. `jq` appears to parse these correctly, where `conftest` does not. 

```sh
$ jq '.' testBOM.json
{
    "test":  "test"
}

$ xxd testBOM.json
00000000: efbb bf7b 0d0a 2020 2020 2274 6573 7422  ...{..    "test"
00000010: 3a20 2022 7465 7374 220d 0a7d 0d0a       :  "test"..}..

$ conftest verify -p foo.rego -d testBOM.json
Error: running verification: load: load documents: 1 error occurred during loading: testBOM.json: invalid character 'ï' looking for beginning of value

$ jq '.' foo.json
{
  "foo": "bar"
}

$ xxd foo.json
00000000: 7b0a 2020 2266 6f6f 223a 2022 6261 7222  {.  "foo": "bar"
00000010: 0a7d 0a                                  .}.

$ conftest verify -p *.rego -d foo.json

1 test, 1 passed, 0 warnings, 0 failures, 0 exceptions, 0 skipped
```

Looking deeper, this appears to be thrown by Go's JSON parser, so not related to conftest itself.

```sh
$ cat foo.go
package main

import (
  "fmt"
  "os"

  "encoding/json"
)

func main() {
  bytes, err := os.ReadFile("testBOM.json")
  if err != nil {
    panic(err)
  }
  j := make(map[string]any)
  if err := json.Unmarshal(bytes, j); err != nil {
    panic(err)
  }
  fmt.Printf("%+v\n", j)
}

$ go run foo.go
panic: invalid character 'ï' looking for beginning of value

goroutine 1 [running]:
main.main()
	/Users/nikon/Downloads/foo.go:17 +0xc4
exit status 2
```

@jinweili1 At this point, I think you can update your JSON input to not have the extra bytes, or file a bug upstream with the Go maintainers.

--- a/parser/json/json.go
+++ b/parser/json/json.go
@@ -10,6 +10,10 @@ type Parser struct{}
 
 // Unmarshal unmarshals JSON files.
 func (p *Parser) Unmarshal(data []byte, v interface{}) error {
+	if len(data) > 2 && data[0] == 0xef && data[1] == 0xbb && data[2] == 0xbf {
+		data = data[3:] // Strip UTF-8 BOM, see https://www.rfc-editor.org/rfc/rfc8259#section-8.1
+	}
+
 	if err := json.Unmarshal(data, v); err != nil {
 		return fmt.Errorf("unmarshal json: %w", err)
 	}

--- a/parser/json/json_test.go
+++ b/parser/json/json_test.go
@@ -40,3 +40,45 @@ func TestJSONParser(t *testing.T) {
 		t.Error("there should be at least one item defined in the parsed file, but none found")
 	}
 }
+
+func TestJSONParserWithBOM(t *testing.T) {
+	tests := []struct {
+		name    string
+		input   []byte
+		want    map[string]any
+		wantErr bool
+	}{
+		{
+			name:  "valid JSON with BOM",
+			input: append([]byte{0xEF, 0xBB, 0xBF}, []byte(`{"test": "value"}`)...),
+			want:  map[string]any{"test": "value"},
+		},
+		{
+			name:  "valid JSON without BOM",
+			input: []byte(`{"test": "value"}`),
+			want:  map[string]any{"test": "value"},
+		},
+	}
+
+	parser := &Parser{}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var got interface{}
+			err := parser.Unmarshal(tt.input, &got)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("Unmarshal() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !tt.wantErr && got == nil {
+				t.Fatal("expected parsed content, got nil")
+			}
+			if m, ok := got.(map[string]any); ok {
+				for k, want := range tt.want {
+					if got := m[k]; got != want {
+						t.Errorf("key %q = %v, want %v", k, got, want)
+					}
+				}
+			}
+		})
+	}
+}

